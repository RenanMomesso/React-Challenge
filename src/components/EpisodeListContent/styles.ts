import styled from "styled-components";
import media from "styled-media-query";

export const EpisodesContent = styled.div`
  width: 90%;
  margin-top: 50px;
  min-height: 400px;
  overflow-y: scroll;
  max-height: 500px;
  ${media.lessThan("medium")`
    max-height: 400px;
    border-bottom: 1px solid lightgray;
    width: 100%;
`}
`;

export const EpisodeSeason = styled.div`
  height: 55px;
  align-items: flex-start;
  justify-content: flex-start;
  display: flex;
`;

export const TableEpisodes = styled.table`
  font-family: Arial, Helvetica, sans-serif;
  border-collapse: collapse;
  width: 100%;
  overflow-y: scroll;
  margin-bottom: 20px;

  td,
  th {
    border: 1px solid #ddd;
    padding: 8px;
    text-align: left;
  }
  th {
    padding-top: 12px;
    padding-bottom: 12px;
    text-align: left;
    background-color: #04aa6d;
    color: white;
  }
`;

export const ButtonSeason = styled.button<{
  seasonNumber: number;
  season: number;
}>`
  height: 100%;
  cursor: pointer;
  width: 85px;
  border-top-right-radius: 20px;
  border: none;
  background-color: ${({ season, seasonNumber }) =>
    seasonNumber === season ? "#04aa6d" : "lightgray"};
  z-index: 10;
  color: ${({ season, seasonNumber }) =>
    seasonNumber === season ? "white" : "black"};
  margin-bottom: -2px;
  transition: all 0.4s ease-in;
`;
