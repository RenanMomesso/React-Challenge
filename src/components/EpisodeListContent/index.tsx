import { useState } from "react";
import EpisodeList from "../Episode";

import * as S from "./styles";

export interface EpisodeListProps {
  episodes: any;
}

const EpisodeListContent = ({ episodes }: EpisodeListProps) => {
  const [seasonNumber, setSeasonNumber] = useState(1);

  return (
    <S.EpisodesContent>
      <S.EpisodeSeason>
        <S.ButtonSeason
          seasonNumber={1}
          season={seasonNumber}
          onClick={() => setSeasonNumber(1)}
        >
          Season 1
        </S.ButtonSeason>
        <S.ButtonSeason
          seasonNumber={2}
          season={seasonNumber}
          onClick={() => setSeasonNumber(2)}
        >
          Season 2
        </S.ButtonSeason>
        <S.ButtonSeason
          seasonNumber={3}
          season={seasonNumber}
          onClick={() => setSeasonNumber(3)}
        >
          Season 3
        </S.ButtonSeason>
      </S.EpisodeSeason>

      <S.TableEpisodes id="customers">
        <thead>
          <tr>
            <th>Season</th>
            <th>Number</th>
            <th>Title</th>
            <th>Date</th>
          </tr>
        </thead>
        {episodes.loading === false &&
          episodes.data.map((item: any) => {
            return (
              item.season === seasonNumber && (
                <EpisodeList
                  key={item.name}
                  season={item.season}
                  serieDate={item.airdate!}
                  serieName={item.name}
                  serieNumber={item.number}
                />
              )
            );
          })}
      </S.TableEpisodes>
    </S.EpisodesContent>
  );
};

export default EpisodeListContent;
