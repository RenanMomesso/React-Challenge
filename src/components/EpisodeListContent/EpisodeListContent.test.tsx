import { screen, render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

import EpisodeListContent from ".";

const EpisodeListContentMock = [
  {
    serieName: "Monster Attack 2",
    season: 1,
    serieDate: "10/10/2016",
    serieNumber: 2,
  },
];

describe("Render EpisodeListContent", () => {
  it("should render EpisodeListContent correctly", () => {
    const {container} = render(
      <BrowserRouter>
        <EpisodeListContent episodes={EpisodeListContentMock} />
      </BrowserRouter>
    );

    expect(container).toMatchSnapshot()
    expect(
      screen.getByRole("button", { name: /season 1/i })
    ).toBeInTheDocument();
    expect(
      screen.getByRole("button", { name: /season 2/i })
    ).toBeInTheDocument();
    expect(
      screen.getByRole("button", { name: /season 3/i })
    ).toBeInTheDocument();
  });
});
