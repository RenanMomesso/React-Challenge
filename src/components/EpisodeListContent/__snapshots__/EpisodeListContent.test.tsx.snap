// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render EpisodeListContent should render EpisodeListContent correctly 1`] = `
<div>
  <div
    class="sc-bdvvtL gKYDZO"
  >
    <div
      class="sc-gsDKAQ iIXhxU"
    >
      <button
        class="sc-hKwDye ilPFjb"
      >
        Season 1
      </button>
      <button
        class="sc-hKwDye gxhiOd"
      >
        Season 2
      </button>
      <button
        class="sc-hKwDye gxhiOd"
      >
        Season 3
      </button>
    </div>
    <table
      class="sc-dkPtRN SSOmN"
      id="customers"
    >
      <thead>
        <tr>
          <th>
            Season
          </th>
          <th>
            Number
          </th>
          <th>
            Title
          </th>
          <th>
            Date
          </th>
        </tr>
      </thead>
    </table>
  </div>
</div>
`;
